unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/TextureHelpers/TextureHelpers.cs

```diff
@@ -68,14 +68,22 @@ public static Texture2D Resize(Texture2D source, int newWidth, int newHeight, bo
 
         return nTex;
     }
-    
+
     public static Texture2D CopyTexture(Texture2D sourceTexture)
     {
-        Texture2D texture = new Texture2D(sourceTexture.width, sourceTexture.height, sourceTexture.format, false);
-        
-        // Note: Surprisingly this works in WebGL here but it doesn't work in Resize()
-        Graphics.CopyTexture(sourceTexture, texture);
-        
-        return texture;
+        Texture2D destTexture = new Texture2D(sourceTexture.width, sourceTexture.height, sourceTexture.format, false);
+
+        if (sourceTexture.mipmapCount > 1)
+        {
+            // Copy the source texture with mips to the destination texture without mips
```

Comment:
Are we discarding mipmaps? Mip maps are extremely important for rendering performance in 3D, not so much in 2D, but it is a must for 3D.